api/docs/reference/search_algorithm.md

```diff
@@ -0,0 +1,226 @@
+[stemming](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/stemming.html).
+When Elasticsearch performs a full text search, it is searching the derived
+tokens, for which it has created quickly searchable index, rather than the
+original text itself. This means that the text analysis configuration applied
+has a significant impact on the way documents are searched and how relevance is
+calculated.
+
+Openverse only applies our custom text analysis configuration to the "title",
+"description", and "tags" fields. All other fields use the default analyser.
+
+Openverse is currently configured exclusively with English language text
+analysis. This means Openverse does not properly index documents in any language
+other than English. This is a known issue that we hope to be able to address
+soon via Elasticsearch's internationalisation tools.
+
+We primarily use the default English language stemming settings aside from
+[minor changes (GitHub)](https://github.com/cc-archive/cccatalog-api/issues/574#issue-668091876)
+present in the `es_mapping.py` configuration to address a specific issue with
+the "anim" stem. We use the
+[Snowball English stemmer](https://snowballstem.org/algorithms/porter/stemmer.html)
+and
+[Lucene's possessive English stemmer](https://lucene.apache.org/core/8_8_0/analyzers-common/org/apache/lucene/analysis/en/EnglishPossessiveFilter.html).
+
+The rationale for this specific stemmer configuration is not documented aside
+from the "anim" stem issue linked above.
+
+Openverse also applies the `lowercase` text analysis filter making our indexes
+case-insensitive.
+
+## Search execution
+
+There are essentially 2 broad types of "full-text" searches available via the
+Openverse API:
+
+1. General "query" searching, which applies a simple text query search against
+   several, non-configurable fields.
+2. Individual field querying, which enables searching specific fields with
+   independent query terms.
+
+Both of these search types also allow further keyword filtering along the
+following fields:
+
+- Extension
+- Category
+- Length
+- Aspect ratio
+- Size
+- Source
+- License
+
+Source is the only field for which you can currently also specify exclusions.
+
+By default, items marked "mature" are excluded, but these can also be enabled.
+
+See the following API documentation links for descriptions and options for each
+field:
+
+- [Audio search](https://api.openverse.engineering/v1/#operation/audio_search)
+- [Image search](https://api.openverse.engineering/v1/#operation/image_search)
+
+Each of these fields are searched relatively strictly, primarily because the
+search domain in each is very small and "keyword" like. That is, there is a
+limited and specific set of terms that appear for the relevant document fields
+for each of these query parameters. All of them are validated to only allow
+specific options (documented in the API documentation links above), which
+enforces the "keyword" like nature of their usage.
```

Comment:
For example, the aspect ratio has the following set of possible values: `square`, `tall`, and `wide`. If you filter by "tall" aspect ratio, you will only get the results that are "tall"; all of the "square" and "wide" items will be discarded from the search results.

Reply:
An example like this can help understand what "keyword" filtering is.

Reply:
Hopefully addressed by the additional term I added. Let me know if not.